trieste/acquisition/rule.py

```diff
@@ -200,14 +202,144 @@ def acquire(
                 self._num_query_points - 1
             ):  # greedily allocate remaining batch elements
                 self._acquisition_function = self._builder.update_acquisition_function(
-                    self._acquisition_function, datasets, models, pending_points=points
+                    self._acquisition_function,
+                    datasets,
+                    models,
+                    pending_points=points,
+                    new_optimization_step=False,
                 )
                 chosen_point = self._optimizer(search_space, self._acquisition_function)
                 points = tf.concat([points, chosen_point], axis=0)
 
         return points
 
 
+class AsyncEGO(AcquisitionRule[State[Optional["AsyncEGO.State"], TensorType], SP_contra]):
+    """AsyncEGO rule, as name suggests, is designed for async BO scenarios.
+    By async BO we understand a use case when multiple objective function can be launched in parallel
+    and are expected to arrive at different times. Instead of waiting for the rest of observations to return,
+    we want to immediately use acquisition function to launch a new observation and avoid wasting computational resources.
+
+    To make the best decision about next point to observe, acquisition function needs to be aware of currently running observations.
+    We call such points "pending". AsyncEGO provides a ``AsyncEGO.State`` object that keeps track of pending points
+    """
+    @dataclass(frozen=True)
+    class State:
```

Comment:
What's the reasoning behind using a frozen dataclass for the state, rather than just the pending_points/outstanding_points?